src/libAtomVM/nifs.c

```diff
@@ -2466,3 +2491,281 @@ static term nif_console_print(Context *ctx, int argc, term argv[])
     }
     return OK_ATOM;
 }
+
+static char b64_table[64] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+// per https://tools.ietf.org/rfc/rfc4648.txt
+
+static term base64_encode(Context *ctx, int argc, term argv[], bool return_binary)
+{
+    UNUSED(argc);
```

Comment:
memory_ensure_free should be called here, so we don't have to worry about src later.

Reply:
Please review changes.